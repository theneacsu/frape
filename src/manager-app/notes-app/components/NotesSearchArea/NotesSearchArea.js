import React from 'react'
import { connect } from 'react-redux'
import { setSearchText } from '../../../../actions/notes-app/filters'
import styles from './NotesSearchArea.module.css'

const NotesSearchArea = props => (
  <div>
    <input
      className={['input', styles.input].join(' ')}
      type="text"
      placeholder="Search in notes"
      autoFocus
      value={props.searchTerm}
      onChange={e => props.setSearchText(e.target.value)}
    />
  </div>
)

const mapDispatchToProps = dispatch => ({
  setSearchText: searchTerm => dispatch(setSearchText(searchTerm))
})

const mapStateToProps = state => ({
  searchTerm: state.notesFilters.searchTerm
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NotesSearchArea)
