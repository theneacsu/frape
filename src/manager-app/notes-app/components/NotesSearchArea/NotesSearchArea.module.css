.div {
  flex: 1;
  display: flex;
}

.input {
  width: 100%;
}

@media (min-width: 450px) {
  .div {
    width: 50%;
    margin-left: 1rem;
  }

  .input {
    height: 5rem;
  }
}
