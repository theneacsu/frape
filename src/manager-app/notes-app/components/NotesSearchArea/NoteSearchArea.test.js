import React from 'react'
import { shallow } from 'enzyme'
import { NotesSearchArea } from './NotesSearchArea'

let searchTerm, setSearchTextSpy, wrapper

beforeEach(() => {
  searchTerm = ''
  setSearchTextSpy = jest.fn()
  wrapper = shallow(
    <NotesSearchArea searchTerm={searchTerm} setSearchText={setSearchTextSpy} />
  )
})

test('Should render NoteSearchArea component correctly with empty searchTerm', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Should render NoteSearchArea component correctly with searchTerm provided', () => {
  searchTerm = 'pizza'
  const wrapper = shallow(
    <NotesSearchArea searchTerm={searchTerm} setSearchText={() => {}} />
  )
  expect(wrapper).toMatchSnapshot()
})

test('Should call setSearchArea with the right value', () => {
  const value = 'coffee'
  wrapper.find('input').simulate('change', {
    target: {
      value
    }
  })
  expect(setSearchTextSpy).toHaveBeenLastCalledWith(value)
})
