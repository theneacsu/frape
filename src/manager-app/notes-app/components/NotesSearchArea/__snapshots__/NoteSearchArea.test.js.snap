// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NoteSearchArea component correctly with empty searchTerm 1`] = `
<div>
  <input
    className="input input"
    onChange={[Function]}
    placeholder="Search in notes"
    type="text"
    value=""
  />
</div>
`;

exports[`Should render NoteSearchArea component correctly with searchTerm provided 1`] = `
<div>
  <input
    className="input input"
    onChange={[Function]}
    placeholder="Search in notes"
    type="text"
    value="pizza"
  />
</div>
`;
